src/models/CommonModel.ts

```diff
@@ -25,4 +25,74 @@ export class CommonModel extends CommonSchema<CommonModel>{
         }
         return newCommonModel;
     }
+
+
+    /**
+     * Only merge if left side is undefined and right side is sat OR both sides are defined
+     * 
+     * @param mergeTo CommonModel to merge into
+     * @param mergeFrom CommonModel to merge values from
+     */
+    static mergeCommonModels(mergeTo: CommonModel, mergeFrom: CommonModel, originalSchema: Schema) : CommonModel {
+        const mergeToProperties = mergeTo.properties;
+        const mergeFromProperties = mergeFrom.properties;
+        if(mergeToProperties === undefined && mergeFromProperties !== undefined){
```

Comment:
I think that logic like:

```ts
if (mergeFromProperties !== undefined) {
  if (mergeToProperties === undefined) {
     mergeTo.properties = mergeFromProperties;
  } else {
     ...
  }
}
```

is easier to read :) But it's only a suggestion.

Reply:
Yea I agree :)